be/src/vec/aggregate_functions/aggregate_function_window.h

```diff
@@ -274,6 +274,8 @@ struct LeadLagData {
                 const auto* nullable_column = assert_cast<const ColumnNullable*>(column);
                 if (nullable_column->is_null_at(0)) {
                     _default_value.reset();
+                } else {
+                    _default_value.set_value(nullable_column->get_nested_column_ptr(), 0);
```

Comment:
seems the default value must be const, so if it's nullable_column, the default value must be NULL?

Reply:
new nereids planner may add a cast to literal string internally like:
`select id, create_time, lead(create_time, 1, '2022-09-06 00:00:00') over
                          (order by create_time desc) as "prev_time" from test1; `
->
`select id, create_time, lead(create_time, 1, cast('2022-09-06 00:00:00' as datetime)) over
                          (order by create_time desc) as "prev_time" from test1; `

so, the above code is added for compatibility of nereids planner